Detekt 1.22.0-RC2 UnusedImports with type resolution false positive for androidx.core

This was not happening in 1.22.0-RC1

When running `detektDebug` in my Android project, I'm getting a bunch of `UnusedImports` false positives only for imports from `androidx.core.*` 

For example:

```
androidx.core.content.getSystemService
androidx.core.content.res.use
androidx.core.view.postDelayed
androidx.core.content.edit
```

Repro project linked above. Steps:

 * Clone https://github.com/TWiStErRob/repros
 * open `detekt/UnusedImports-androidx.core-rc`
 * `gradlew cleanDetektDebug detektDebug`

Expected:
```
src\main\java\com\example\layoutInflater.kt:8:11: Calling !! on a nullable type will throw a NullPointerException at runtime in case the value is null. It should be avoided. [UnsafeCallOnNullableType]
```

Actual:
```
src\main\java\com\example\layoutInflater.kt:5:17: error: unresolved reference: core
import androidx.core.content.getSystemService
                ^
src\main\java\com\example\layoutInflater.kt:8:11: error: type mismatch: inferred type is Any! but LayoutInflater? was expected
  get() = getSystemService()!!
          ^
src\main\java\com\example\layoutInflater.kt:8:28: error: no value passed for parameter 'p0'
  get() = getSystemService()!!
                           ^

src\main\java\com\example\layoutInflater.kt:5:1: The import 'androidx.core.content.getSystemService' is unused. [UnusedImports]
```

@detekt this looks like a regression, because with the repro it's easy to confirm that it was indeed working in RC1. And it seems to highlight a bigger issue than just `UnusedImports`.

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt
@@ -208,8 +208,8 @@ abstract class Detekt @Inject constructor(
     }
 
     @get:Internal
-    internal val arguments: Provider<List<String>> = project.provider {
-        listOf(
+    internal val arguments
+        get() = listOf(
             InputArgument(source),
             ClasspathArgument(classpath),
             LanguageVersionArgument(languageVersionProp.orNull),
@@ -230,7 +230,6 @@ abstract class Detekt @Inject constructor(
             BasePathArgument(basePathProp.orNull),
             DisableDefaultRuleSetArgument(disableDefaultRuleSetsProp.getOrElse(false))
         ).plus(convertCustomReportsToArguments()).flatMap(CliArgument::toArgument)
-    }
 
     @InputFiles
     @SkipWhenEmpty
@@ -252,7 +251,7 @@ abstract class Detekt @Inject constructor(
     @TaskAction
     fun check() {
         DetektInvoker.create(task = this, isDryRun = isDryRun).invokeCli(
-            arguments = arguments.get(),
+            arguments = arguments,
             ignoreFailures = ignoreFailures,
             classpath = detektClasspath.plus(pluginClasspath),
             taskName = name

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektCreateBaselineTask.kt
@@ -20,7 +20,6 @@ import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.FileTree
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.provider.Property
-import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.CacheableTask
 import org.gradle.api.tasks.Classpath
 import org.gradle.api.tasks.Console
@@ -115,8 +114,8 @@ abstract class DetektCreateBaselineTask : SourceTask() {
     abstract val jdkHome: DirectoryProperty
 
     @get:Internal
-    internal val arguments: Provider<List<String>> = project.provider {
-        listOf(
+    internal val arguments
+        get() = listOf(
             CreateBaselineArgument,
             ClasspathArgument(classpath),
             JvmTargetArgument(jvmTargetProp.orNull),
@@ -131,7 +130,6 @@ abstract class DetektCreateBaselineTask : SourceTask() {
             BasePathArgument(basePathProp.orNull),
             DisableDefaultRuleSetArgument(disableDefaultRuleSets.getOrElse(false))
         ).flatMap(CliArgument::toArgument)
-    }
 
     @InputFiles
     @SkipWhenEmpty
@@ -142,7 +140,7 @@ abstract class DetektCreateBaselineTask : SourceTask() {
     @TaskAction
     fun baseline() {
         DetektInvoker.create(task = this).invokeCli(
-            arguments = arguments.get(),
+            arguments = arguments,
             ignoreFailures = ignoreFailures.getOrElse(false),
             classpath = detektClasspath.plus(pluginClasspath),
             taskName = name

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektGenerateConfigTask.kt
@@ -10,7 +10,6 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.model.ObjectFactory
-import org.gradle.api.provider.Provider
 import org.gradle.api.services.BuildService
 import org.gradle.api.services.BuildServiceParameters
 import org.gradle.api.tasks.CacheableTask
@@ -57,12 +56,11 @@ abstract class DetektGenerateConfigTask @Inject constructor(
         }
 
     @get:Internal
-    internal val arguments: Provider<List<String>> = project.provider {
-        listOf(
+    internal val arguments
+        get() = listOf(
             GenerateConfigArgument,
             ConfigArgument(configFile.get())
         ).flatMap(CliArgument::toArgument)
-    }
 
     @TaskAction
     fun generateConfig() {
@@ -74,7 +72,7 @@ abstract class DetektGenerateConfigTask @Inject constructor(
         Files.createDirectories(configFile.get().asFile.parentFile.toPath())
 
         DetektInvoker.create(task = this).invokeCli(
-            arguments = arguments.get(),
+            arguments = arguments,
             classpath = detektClasspath.plus(pluginClasspath),
             taskName = name,
         )

--- a/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/internal/DetektAndroid.kt
+++ b/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/internal/DetektAndroid.kt
@@ -16,7 +16,6 @@ import io.gitlab.arturbosch.detekt.extensions.DetektExtension
 import org.gradle.api.DomainObjectSet
 import org.gradle.api.Project
 import org.gradle.api.Task
-import org.gradle.api.attributes.Attribute
 import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.FileCollection
 import org.gradle.api.tasks.TaskProvider
@@ -123,11 +122,7 @@ internal fun Project.registerAndroidDetektTask(
         setSource(variant.sourceSets.map { it.javaDirectories + it.kotlinDirectories })
         extraInputSource?.let { source(it) }
         classpath.setFrom(
-            variant.compileConfiguration.incoming.artifactView { view ->
-                view.attributes {
-                    it.attribute(Attribute.of("artifactType", String::class.java), "jar")
-                }
-            }.files,
+            variant.getCompileClasspath(null).filter { it.exists() },
             bootClasspath,
             javaCompileDestination(variant),
         )
@@ -151,11 +146,7 @@ internal fun Project.registerAndroidCreateBaselineTask(
         setSource(variant.sourceSets.map { it.javaDirectories + it.kotlinDirectories })
         extraInputSource?.let { source(it) }
         classpath.setFrom(
-            variant.compileConfiguration.incoming.artifactView { view ->
-                view.attributes {
-                    it.attribute(Attribute.of("artifactType", String::class.java), "jar")
-                }
-            }.files,
+            variant.getCompileClasspath(null).filter { it.exists() },
             bootClasspath,
             javaCompileDestination(variant),
         )

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektJvmSpec.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektJvmSpec.kt
@@ -37,7 +37,7 @@ class DetektJvmSpec {
         val project = gradleRunner.buildProject()
 
         val detektTask = project.tasks.getByPath("detektMain") as Detekt
-        val argumentString = detektTask.arguments.get().joinToString(" ")
+        val argumentString = detektTask.arguments.joinToString(" ")
 
         assertThat(argumentString).containsPattern("""--baseline \S*[/\\]detekt-baseline-main.xml """)
         assertThat(argumentString).contains("--report xml:")
@@ -53,7 +53,7 @@ class DetektJvmSpec {
         val project = gradleRunner.buildProject()
 
         val detektTask = project.tasks.getByPath("detektTest") as Detekt
-        val argumentString = detektTask.arguments.get().joinToString(" ")
+        val argumentString = detektTask.arguments.joinToString(" ")
 
         assertThat(argumentString).containsPattern("""--baseline \S*[/\\]detekt-baseline-test.xml """)
         assertThat(argumentString).contains("--report xml:")

--- a/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektPlainSpec.kt
+++ b/detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektPlainSpec.kt
@@ -68,7 +68,7 @@ class DetektPlainSpec {
             val project = gradleRunner.buildProject()
 
             val detektTask = project.tasks.getByPath("detekt") as Detekt
-            val argumentString = detektTask.arguments.get().joinToString(" ")
+            val argumentString = detektTask.arguments.joinToString(" ")
 
             assertThat(argumentString).containsPattern("""--baseline \S*[/\\]detekt-baseline.xml """)
             assertThat(argumentString).contains("--report xml:")

